types/async-busboy/async-busboy-tests.ts

```diff
@@ -0,0 +1,16 @@
+import * as Koa from 'koa';
+import * as asyncBusboy from 'async-busboy';
```

Comment:
Since the export is commonjs-style `export = asyncBusboy`, the import should be commonjs-style `import asyncBusboy = require('async-busboy')`

Reply:
Done. I've also checked all import statements.

Reply:
hmm, I'm confused.
https://www.typescriptlang.org/docs/handbook/modules.html surely said:

> When exporting a module using export =, TypeScript-specific import module = require("module") must be used to import the module.

But, not all packages in this repository follow that suggestions... what's the difference practically?

Reply:
That's the default pattern. You can vary from it if you know what you're doing, for example if you export= an object literal, you can import it with `import * as xxx from 'yyy'` because the object literal isn't callable, and the ES module spec says that `* as xxx` results in a non-callable `xxx`. But generally, it's simpler if those two syntaxes always go together.

Reply:
Thanks. Could you give me some advices?
I understand the reason that I cannot do `import * as xxx from 'yyy'` for callable.
```
declare function xxx(): void;

export = xxx;
```

However, it looks like there are some tricks that skip compile errors.

```
declare function xxx(): void;
namespace xxx {};

export = xxx;
```

```
interface XXX {
  (): void;
}

var xxx: XXX;

export = xxx;
```

According to your suggestion, I feel that these tricks are illegal and the `tsc` should fail in the future.
On the other hand, DefinitelyTyped have removed the notice about the first trick. https://github.com/DefinitelyTyped/DefinitelyTyped/commit/2379ab7be2142b750ae2615e30f72638026127b2

..So, are these tricks legitimate in today? and why?

Thanks, again.